executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Ledger.Spec.STS.Chain.Transactions where
+
+import           Cardano.Prelude (HeapWords, heapWords, heapWords1)
+import           Data.Bimap (Bimap)
+import           Data.Function ((&))
+import           Hedgehog (Gen)
+import qualified Hedgehog.Gen as Gen
+import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
+                     GenericSemigroup (GenericSemigroup))
+import           GHC.Generics (Generic)
+
+import           Control.State.Transition (Embed, Environment, PredicateFailure,
+                     STS, Signal, State, TRC (TRC), initialRules,
+                     judgmentContext, trans, transitionRules, wrapFailed)
+import           Control.State.Transition.Generator (sigGen, genTrace)
+import           Control.State.Transition.Trace (traceSignals, TraceOrder(OldestFirst))
+
+import           Ledger.Core (Slot)
+import qualified Ledger.Core as Core
+
+import           Cardano.Ledger.Spec.STS.Sized (WordCount, size)
+import qualified Cardano.Ledger.Spec.STS.Dummy.Transaction as Dummy
+import           Cardano.Ledger.Spec.STS.Update.Ideation (IDEATION)
+import qualified Cardano.Ledger.Spec.STS.Update.Ideation as Ideation
+
+
+data TRANSACTIONS
+
+
+data Env =
+  Env { currentSlot :: Slot
+      , participants :: Bimap Core.VKey Core.SKey
+      }
+  deriving (Eq, Show)
+
+
+data St =
+  St { dummySt :: State Dummy.TRANSACTION
+     , ideationSt :: State IDEATION
```

Comment:
We do not store some state that can give us the slot of an Ideation Signal. We only have a `Map` from Dummy Transactions to Slots. We need to be able to get fast, the slot of a specific update event, in order to do slot arithmetic.

Reply:
Maybe we need to add in the Ideation state the following
```haskell
Map (Signal IDEATION) Slot
```